src/shared/sam/debugger/awsSamDebugConfigurationValidator.ts

```diff
@@ -12,96 +13,120 @@ import {
-    public constructor(private readonly cftRegistry = CloudFormationTemplateRegistry.getRegistry()) {}
-
-    public validateSamDebugConfiguration(debugConfiguration: AwsSamDebuggerConfiguration): ValidationResult {
-        const generalValidationResult = this.generalDebugConfigValidation(debugConfiguration)
-        if (!generalValidationResult.isValid) {
-            return generalValidationResult
+    public constructor(
+        private readonly cftRegistry = CloudFormationTemplateRegistry.getRegistry(),
+        private readonly workspaceFolder: vscode.WorkspaceFolder | undefined
+    ) {}
+
+    /**
+     * Validates debug configuration properties.
+     */
+    public validate(config: AwsSamDebuggerConfiguration): boolean {
+        let rv: { isValid: boolean; message?: string } = { isValid: false, message: undefined }
+        if (!config.request) {
+            rv.message = localize(
+                'AWS.sam.debugger.missingField',
+                'Missing required field "{0}" in debug config',
+                'request'
+            )
+        } else if (!AWS_SAM_DEBUG_REQUEST_TYPES.includes(config.request)) {
+            rv.message = localize(
+                'AWS.sam.debugger.invalidRequest',
+                'Debug Configuration has an unsupported request type. Supported types: {0}',
+                AWS_SAM_DEBUG_REQUEST_TYPES.join(', ')
+            )
+        } else if (!AWS_SAM_DEBUG_TARGET_TYPES.includes(config.invokeTarget.target)) {
+            rv.message = localize(
+                'AWS.sam.debugger.invalidTarget',
+                'Debug Configuration has an unsupported target type. Supported types: {0}',
+                AWS_SAM_DEBUG_TARGET_TYPES.join(', ')
+            )
+        } else if (config.invokeTarget.target === TEMPLATE_TARGET_TYPE) {
+            let cfnTemplate
+            if (config.invokeTarget.samTemplatePath) {
+                const fullpath = tryGetAbsolutePath(this.workspaceFolder, config.invokeTarget.samTemplatePath)
+                // Normalize to absolute path for use in the runner.
+                config.invokeTarget.samTemplatePath = fullpath
+                cfnTemplate = this.cftRegistry.getRegisteredTemplate(fullpath)?.template
+            }
+            rv = this.validateTemplateConfig(config, config.invokeTarget.samTemplatePath, cfnTemplate)
+        } else if (config.invokeTarget.target === CODE_TARGET_TYPE) {
+            rv = this.validateCodeConfig(config)
         }
 
-        if (isTemplateTargetProperties(debugConfiguration.invokeTarget)) {
-            return this.templateDebugConfigValidation(debugConfiguration, this.cftRegistry)
-        } else if (isCodeTargetProperties(debugConfiguration.invokeTarget)) {
-            return this.codeDebugConfigValidation(debugConfiguration)
+        if (!rv.isValid) {
+            getLogger().error(`invalid debug-config: ${rv.message}`)
+            vscode.window.showErrorMessage(rv.message ?? 'invalid debug-config')
```

Comment:
"invalid debug-config" is not a helpful message for users. If the only way this state can happen is through an error on our part, this should throw an error, so we know during extension development that the code is incorrect.

Reply:
SGTM. I've seen in other places we are reluctant to throw errors.

Reply:
We haven't been throwing Errors with consistency or an appropriate strategy. Definitely need to improve that moving forward.